packages/client/src/client/index.ts

```diff
@@ -0,0 +1,191 @@
+import axios, { AxiosError, AxiosInstance } from 'axios';
+import axiosRetry, { exponentialDelay, IAxiosRetryConfig } from 'axios-retry';
+import { StatusCodes } from 'http-status-codes';
+import { stringify } from 'qs';
+import { ILogger, KitMetadata, TileDetails, TileDetailsPayload, TileParams, TileParamsWithKit, TileQueryParams } from '@map-colonies/detiler-common';
+import { DEFAULT_PAGE_SIZE, DEFAULT_RETRY_STRATEGY_DELAY, DetilerClientConfig, DetilerOptions, RetryStrategy } from './config';
+
+export interface IDetilerClient {
+  getKits: () => Promise<KitMetadata[]>;
+  queryTilesDetails: (params: TileQueryParams) => Promise<TileDetails[]>;
+  queryTilesDetailsAsyncGenerator: (params: TileQueryParams) => AsyncGenerator<TileDetails[]>;
+  getTileDetails: (params: TileParamsWithKit) => Promise<TileDetails | null>;
+  getTilesDetails: (params: TileParams & { kits?: string[] }) => Promise<TileDetails[]>;
+  setTileDetails: (params: TileParamsWithKit, payload: TileDetailsPayload) => Promise<void>;
+}
+
+export class DetilerClient implements IDetilerClient {
+  private readonly logger: ILogger | undefined;
+  private readonly config: DetilerClientConfig;
+  private readonly axios: AxiosInstance;
+
+  public constructor(options: DetilerOptions) {
+    const { logger, ...config } = options;
+    this.logger = logger;
+    this.config = config;
+    this.axios = axios.create({ timeout: options.timeout });
```

Comment:
I think the backend is going to be protected with an api-key. If so, you should add here the header.

Reply:
done.